executor/runtime/docker/docker.go

```diff
@@ -2047,20 +2047,27 @@ func (r *DockerRuntime) handleDockerEvent(message events.Message, statusMessageC
 		l.Debugf("Skipping docker start event for %s, no need to update the pod", cName)
 		return nonTerminalDockerEvent
 	case "die":
-		// TODO: Handle the difference between platform/user sidecar, not all
-		// "die"s should result in a Finish
-		if exitCode := message.Actor.Attributes["exitCode"]; exitCode == "0" {
+		exitCode := message.Actor.Attributes["exitCode"]
+		if exitCode == "0" {
 			statusMessageChan <- runtimeTypes.StatusMessage{
 				Status: runtimeTypes.StatusFinished,
 				Msg:    cName + " container successfully exited with 0",
 			}
+			return isTerminalDockerEvent
+		} else if exitCode == "137" && cName != mainContainerName {
```

Comment:
A lot of us are trained to think `137 = OOMKill`. I am not sure this is the best idea.

Reply:
I agree. :(
Hopefully in those cases the `oom` docker event will come through.

Reply:
Yea we need to retrain to think "137" == killed with signal 9 (128 + 9), which is *usually* killed by oom killer